src/com/sap/piper/integration/CloudFoundry.groovy

```diff
@@ -43,7 +43,7 @@ class CloudFoundry {
     def getAppRefUrl(String apiEndpoint, String org, String space, String bearerToken, String appName, boolean verbose){
         def orgGuid = getOrgGuid(apiEndpoint, org, bearerToken, verbose)
         def spaceGuid = getSpaceGuid(apiEndpoint, orgGuid, space, bearerToken, verbose)
-        def appInfo = script.httpRequest  url:"${apiEndpoint}/v3/apps?names=${appName},${appName}_blue,${appName}_green,${appName}-blue,${appName}-green,space_guids=${spaceGuid}",  quiet: !verbose,
+        def appInfo = script.httpRequest  url:"${apiEndpoint}/v3/apps?names=${appName},${appName}_blue,${appName}_green,${appName}-blue,${appName}-green&space_guids=${spaceGuid}",  quiet: !verbose,
```

Comment:
Not directly related to that fix (which is basically OK): 

* shoulnd't we encode the comma `,`-> `%2C`? Should of course not be done "manually" but via an url encoder.
* [curiosity] why are the ${appName}_blue and ${appName}_green contained twice?

Reply:
> shoulnd't we encode the comma ,-> %2C? Should of course not be done "manually" but via an url encoder.

I would hope that url encoding is done by the httpRequest plugin ...

>[curiosity] why are the ${appName}_blue and ${appName}_green contained twice?
It is not. It once with _ and once with -. I guess the _  could also be removed but was in first.

Reply:
> I would hope

ahh, yes, did not read carefully enought. Would expect the same.

Reply:
1. Good point, I hoped that the `httpRequest` already does it
2. It's in there, once with `-` and once with `_` as a delimiter. From own experience (missing documentation anywhere) the apps are currently deployed in the pipeline with the `-` delimiter. For backwards compatibility I kept the `_` ones in there (#1805).

Reply:
> Good point, I hoped that the httpRequest already does it

As already commented above: my mistake. I would expect the same. In case it wouldn't to that it should be fixed there. But i also assume it is done there ... Sorry for disturbing ...

Reply:
Hey, no worries...my answer was just a bit slow. If I had read Daniels answer above I wouldn't have posted at all^^